### PR TITLE
Core support for external Vault Support for Carbon Configurations

### PR DESCRIPTION
**Pull request by sanduniNagahage**
$secret { alias }  : $secret { provider: repository: alias }
>The SecretManager and the SecretManagerSecretCallbackHandler classes and the SecretRepository and the SecretRepositoryProvider Interfaces of the Carbon Securevault will be modified to build the required core support.

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

##Related PRs
>https://github.com/wso2/config-mapper/pull/19

### CODE REVIEW

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 1]
```diff
@@ -214,6 +191,51 @@ public void init(Properties properties) {
         initialized = true;
     }
 
+    /**
+     * Check whether to use the provider listed under secretRepositories
```
Comment (Buddhimah) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:195]: Lets add a fullstop to end of the comment

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 2]
```diff
@@ -214,6 +191,51 @@ public void init(Properties properties) {
...
+                    return getSecret((String) providers.keySet().toArray()[0],
+                            (String) secretRepositories.keySet().toArray()[0],
+                            annotation[0]);
+                case 3:
+                    return getSecret(annotation[0], annotation[1], annotation[2]);
+                default:
+                    throw new IllegalArgumentException("invalid annotation");
+            }
+        } catch (NullPointerException e) {
+            handleException("No Secret Repositories have been initialized : ");
```
Comment (Buddhimah) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:234]:
```suggestion
            handleException("No Secret Repositories have been initialized: ");
```

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 3]
```diff
@@ -214,6 +191,51 @@ public void init(Properties properties) {
...
+        try {
+            switch (length) {
+                case 1:
+                    return getSecret((String) providers.keySet().toArray()[0],
+                            (String) secretRepositories.keySet().toArray()[0],
+                            annotation[0]);
+                case 3:
+                    return getSecret(annotation[0], annotation[1], annotation[2]);
+                default:
+                    throw new IllegalArgumentException("invalid annotation");
```
Comment (Buddhimah) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:231]:
```suggestion
                    throw new IllegalArgumentException("Invalid annotation.");
```

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 4]
```diff
@@ -284,4 +327,190 @@ private boolean validatePasswords(String identityStorePass,
...
+    }
+
+    /**
+     * Terminates if either properties, secretRepositories or secretProviders haven`t been configured
+     */
+    private void validateSecureVaultStatus() {
+
+        if (!(isLegacyProvidersExists || isNovelProvidersExists)) {
+            if (log.isDebugEnabled()) {
+                log.debug("No secret repositories have been configured");
```
Comment (Buddhimah) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:383]:
```suggestion
                log.debug("No secret repositories have been configured.");
```

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 5]
```diff
@@ -284,4 +327,190 @@ private boolean validatePasswords(String identityStorePass,
...
+
+    /**
+     * Validate the property value to avoid the processing of null values
+     *
+     * @param propValue Value of the required property
+     * @return Return true if not null
+     */
+    private boolean validatePropValue(String propValue) {
+
+        if (propValue == null || "".equals(propValue)) {
```
Comment (Buddhimah) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:424]: If possible use StringUtils

File: src/main/java/org/wso2/securevault/secret/SecretRepository.java [hunk 6]
```diff
@@ -25,6 +25,16 @@
...
  */
 public interface SecretRepository {
+    /**
+     * Provide the type of the secret repository
+     *
+     * @return secret repository type
+     */
+    default String getType() {
+
+        return "secret repository type";
```
Comment (Buddhimah) [inline src/main/java/org/wso2/securevault/secret/SecretRepository.java:35]: can we have a different default type

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 7]
```diff
@@ -214,6 +191,51 @@ public void init(Properties properties) {
...
+                case 1:
+                    return getSecret((String) providers.keySet().toArray()[0],
+                            (String) secretRepositories.keySet().toArray()[0],
+                            annotation[0]);
+                case 3:
+                    return getSecret(annotation[0], annotation[1], annotation[2]);
+                default:
+                    throw new IllegalArgumentException("invalid annotation");
+            }
+        } catch (NullPointerException e) {
```
Comment (Buddhimah) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:233]: Cant we do a null check and avoid this

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 2 above)
Comment (Buddhimah) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java:234]: e is not passed to handleException method

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 8]
```diff
@@ -214,6 +191,51 @@ public void init(Properties properties) {
...
+    /**
+     * Resolve the secret annotation for the secrets coming from
+     * repositories belongs to providers listed under secretProviders property
+     *
+     * @param annotation vaule retrieve by the resolveSecret as the value to be resolved
+     * @return If there is a secret , otherwise , alias itself
+     */
+    public String resolveSecret(String[] annotation) {
+
+        int length = annotation.length;
```
Comment (Buddhimah) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:221]: why we need this variable

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 9]
```diff
@@ -230,6 +252,27 @@ public String getSecret(String alias) {
...
+     * @param alias      alias to be resolved
+     * @return If there is a secret , otherwise , alias itself
+     */
+    public String getSecret(String provider, String repository, String alias) {
+
+        if (providers.containsKey(provider) && secretRepositories.containsKey(repository)) {
+            return secretRepositories.get(repository).getSecret(alias);
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("No such secret repository listed under configurations");
```
Comment (Buddhimah) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:270]:
```suggestion
            log.debug("No such secret repository listed under configurations.");
```

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 10]
```diff
@@ -284,4 +327,190 @@ private boolean validatePasswords(String identityStorePass,
...
+    /**
+     * Creates the TrustKeyStoreWrapper and the IdentityKeyStoreWrapper
+     *
+     * @param identityKeyStoreWrapper Represents the private keyStore entry
+     * @param trustKeyStoreWrapper    Represents the abstraction for trusted KeyStore
+     * @param properties              Configuration properties
+     */
+    private void createKeyStoreWrappers(IdentityKeyStoreWrapper identityKeyStoreWrapper,
+                                        TrustKeyStoreWrapper trustKeyStoreWrapper, Properties properties) {
+        //Create a KeyStore Information  for private key entry KeyStore
```
Comment (Buddhimah) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:459]: new line

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 11]
```diff
@@ -185,14 +155,21 @@ public void init(Properties properties) {
...
-                            getSecretRepository(identityKeyStoreWrapper, trustKeyStoreWrapper);
-                    secretRepository.init(configurationProperties, id);
-                    if (parentRepository == null) {
-                        parentRepository = secretRepository;
+                    if (PROP_SECRET_PROVIDERS.equals(propertyName)) {
+                        Properties filteredConfigs = filterConfigurations(providerType, configurationProperties);
+                        HashMap<String, SecretRepository> providerBasedSecretRepositories =
+                                ((SecretRepositoryProvider) instance).initProvider(filteredConfigs, providerType);
+                        secretRepositories.putAll(providerBasedSecretRepositories);
+                    }else {
```
Comment (Buddhimah) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:163]: Formatting

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 8 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java:221]: Use to differentiate the secret annotation
if length == 1 -> provider type and the repository type should be assigned accordingly
if length == 3 -> annotation itself will provide the provider and the repository type

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 12]
```diff
@@ -215,31 +192,97 @@ public void init(Properties properties) {
...
+     * @param annotation vaule retrieve by the resolveSecret as the value to be resolved.
+     * @return If there is a secret , otherwise , alias itself.
+     */
+    public String resolveSecret(String[] annotation) {
+
+        int length = annotation.length;
+        try {
+            switch (length) {
+                case 1:
+                    return getSecret((String) providers.keySet().toArray()[0],
```
Comment (Buddhimah) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:225]: better to add a comment explaining bit about this logic

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 13]
```diff
@@ -215,31 +192,97 @@ public void init(Properties properties) {
...
+            }
+        }
+        return resolveSecret(secretAnnotationStrings);
+    }
+
+    /**
+     * Resolve the secret annotation for the secrets coming from
+     * repositories belongs to providers listed under secretProviders property.
+     *
+     * @param annotation vaule retrieve by the resolveSecret as the value to be resolved.
```
Comment (Buddhimah) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:216]: typo value

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 2 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java:234]: Remove NPE catch and replaced it with a null check

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 1 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java:195]: Done

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 3 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java]: Enhance the error message

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 5 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java:424]: Utilized with StringUtils

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 7 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java:233]: Replaced with a null check

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 13 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java:216]: Corrected

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 11 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java:163]: Already Formatted, What is the correct way though?

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 12 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java:225]: Added a descriptive comment

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 13 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java:216]: Corrected

File: src/main/java/org/wso2/securevault/secret/SecretRepository.java (hunk 6 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretRepository.java:35]: Removed the getType method and instead get the type via the configuration file

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 10 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java]: Brought to a new line

File: output.diff [hunk 14]
```diff
@@ -0,0 +1,22 @@
+diff --git a/src/main/java/org/wso2/securevault/secret/SecretManager.java b/src/main/java/org/wso2/securevault/secret/SecretManager.java
```
Comment (Buddhimah) [inline output.diff:1]: What is this file?

File: src/main/java/org/wso2/securevault/secret/repository/FileBaseSecretRepository.java [hunk 15]
```diff
@@ -84,8 +84,8 @@ public void init(Properties properties, String id) {
 
         Properties cipherProperties = MiscellaneousUtil.loadProperties(filePath);
         if (cipherProperties.isEmpty()) {
-            if (log.isDebugEnabled()) {
-                log.debug("Cipher texts cannot be loaded form : " + filePath);
+            if (log.isInfoEnabled()){
```
Comment (sahanruwanga) [inline src/main/java/org/wso2/securevault/secret/repository/FileBaseSecretRepository.java:87]: this line should be formatted.

File: output.diff (hunk 14 above)
Comment (sanduniNagahage) [reply output.diff:1]: It is a file that contains information about the changes in the file. It has been removed as it is not required

File: src/main/java/org/wso2/securevault/secret/repository/FileBaseSecretRepository.java (hunk 15 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/repository/FileBaseSecretRepository.java:87]: Formatted

File: output.diff [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+index 3c7f7df..9e8ddae 100644
+--- a/src/main/java/org/wso2/securevault/secret/SecretManager.java
++++ b/src/main/java/org/wso2/securevault/secret/SecretManager.java
+@@ -122,6 +122,8 @@ public class SecretManager {
+             }
+         }
+
```
Comment (tharindu1st) [inline output.diff:8]: do we need output.diff ?

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 17]
```diff
@@ -13,51 +13,70 @@
...
+    private String globalSecretProvider = null;
+    // property key for global secret provider.
+    private final static String PROP_SECRET_PROVIDER = "carbon.secretProvider";
 
-    // global password provider implementation class if defined in secret manager conf file
-    private String globalSecretProvider =null;
-    // property key for global secret provider
-    private final static String PROP_SECRET_PROVIDER="carbon.secretProvider";
+    /* Hash map to keep the providers listed under secretRepositories and secretProviders property. */
+    private HashMap<String, String> providers = new HashMap<>();
```
Comment (tharindu1st) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:64]: HashMap -> Map in declaration

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 18]
```diff
@@ -13,51 +13,70 @@
...
+    /* Hash map to keep the secret repositories coming from a provider listed under secretProviders property. */
+    private HashMap<String, SecretRepository> secretRepositories = new HashMap<>();
```
Comment (tharindu1st) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:66]: same goes here

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 19]
```diff
@@ -185,14 +158,21 @@ public void init(Properties properties) {
...
 
                 if (instance instanceof SecretRepositoryProvider) {
-                    SecretRepository secretRepository = ((SecretRepositoryProvider) instance).
-                            getSecretRepository(identityKeyStoreWrapper, trustKeyStoreWrapper);
-                    secretRepository.init(configurationProperties, id);
-                    if (parentRepository == null) {
-                        parentRepository = secretRepository;
+                    if (PROP_SECRET_PROVIDERS.equals(propertyName)) {
+                        Properties filteredConfigs = filterConfigurations(providerType, configurationProperties);
+                        HashMap<String, SecretRepository> providerBasedSecretRepositories =
```
Comment (tharindu1st) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:163]: same goes here

File: src/main/java/org/wso2/securevault/secret/repository/FileBaseSecretRepository.java (hunk 15 above)
Comment (tharindu1st) [inline src/main/java/org/wso2/securevault/secret/repository/FileBaseSecretRepository.java:87]: why do we checking log.info enabled

**Review by tharindu1st — CHANGES_REQUESTED** [review]

File: output.diff (hunk 16 above)
Comment (sanduniNagahage) [reply output.diff:8]: removed

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 17 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java:64]: Changed

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 18 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java:66]: Changed

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 19 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java:163]: Changed

File: src/main/java/org/wso2/securevault/secret/repository/FileBaseSecretRepository.java (hunk 15 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/repository/FileBaseSecretRepository.java:87]: Revert Back to the old implementation -> Debug Logs

File: src/test/java/org/wso2/securevault/secret/SecretManagerTest.java [hunk 20]
```diff
@@ -0,0 +1,195 @@
...
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({SecretRepository.class})
+
+public class SecretManagerTest {
+
+    private SecretManager secretManager;
+
+    @BeforeClass
+    public void steUp() {
```
Comment (Buddhimah) [inline src/test/java/org/wso2/securevault/secret/SecretManagerTest.java:43]:
```suggestion
    public void setUp() {
```

File: src/main/java/org/wso2/securevault/secret/handler/SecretManagerSecretCallbackHandler.java [hunk 21]
```diff
@@ -1,47 +1,53 @@
...
+ */
+public class SecretManagerSecretCallbackHandler extends AbstractSecretCallbackHandler {
+
+    private final SecretManager secretManager = SecretManager.getInstance();
+
+    protected void handleSingleSecretCallback(SingleSecretCallback singleSecretCallback) {
+
+        if (!secretManager.isInitialized()) {
+            if (log.isWarnEnabled()) {
+                log.warn("SecretManager has not been initialized.Cannot collect secrets.");
```
Comment (sahanruwanga) [inline src/main/java/org/wso2/securevault/secret/handler/SecretManagerSecretCallbackHandler.java:37]: Need a space after the end of the first sentence.

**Review by tharindu1st — APPROVED** [review]

File: src/main/java/org/wso2/securevault/secret/handler/SecretManagerSecretCallbackHandler.java [hunk 22]
```diff
@@ -1,47 +1,53 @@
...
+            }
+            return;
+        }
+
+        String id = singleSecretCallback.getId();
+        if (id != null && !"".equals(id)) {
+            if(log.isDebugEnabled()){
+                log.debug("The secret annotation provided : " + id);
```
Comment (sahanruwanga) [inline src/main/java/org/wso2/securevault/secret/handler/SecretManagerSecretCallbackHandler.java:45]: recheck grammer of the full debug log.

File: src/main/java/org/wso2/securevault/secret/handler/SecretManagerSecretCallbackHandler.java [hunk 23]
```diff
@@ -1,47 +1,53 @@
...
+            }
+            singleSecretCallback.setSecret(secretManager.resolveSecret(id));
+        }
+        else {
+            log.error("Secret Annotation provided is empty.");
```
Comment (sahanruwanga) [inline src/main/java/org/wso2/securevault/secret/handler/SecretManagerSecretCallbackHandler.java:50]: Recheck grammer.

File: src/test/java/org/wso2/securevault/secret/SecretManagerTest.java (hunk 20 above)
Comment (sanduniNagahage) [reply src/test/java/org/wso2/securevault/secret/SecretManagerTest.java]: corrected

File: pom.xml [hunk 24]
```diff
@@ -164,7 +157,12 @@
             <version>6.11</version>
             <scope>test</scope>
         </dependency>
-
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-core</artifactId>
+            <version>1.7.1</version>
```
Comment (Buddhimah) [inline pom.xml:163]: This is not the practise that we provide the check other places

File: src/main/java/org/wso2/securevault/secret/SecretManager.java [hunk 25]
```diff
@@ -215,53 +194,150 @@ public void init(Properties properties) {
...
+        if (secretAnnotationStrings.length == 1) {
+            if (isLegacyProvidersExists) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Getting secrets from legacy provider.");
+                }
+                return getSecret(secretAnnotation);
+            }
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("Getting secrets from a provider other than the legacy provider. ");
```
Comment (Buddhimah) [inline src/main/java/org/wso2/securevault/secret/SecretManager.java:215]:
```suggestion
            log.debug("Getting secrets from a provider other than the legacy provider.");
```

File: src/test/java/org/wso2/securevault/secret/SecretManagerTest.java [hunk 26]
```diff
@@ -0,0 +1,156 @@
...
+
+    @Test(description = "Test case for filterConfigurations() method. ")
+    public void testFilterConfigurations() throws Exception {
+
+        Properties actual = Whitebox.invokeMethod(secretManager, "filterConfigurations", "secretProviders",
+                getConfigProperties());
+        Assert.assertEquals(12, actual.size());
+    }
+
+    @Test(description = "Test case for addStringToArray() method. ")
```
Comment (Buddhimah) [inline src/test/java/org/wso2/securevault/secret/SecretManagerTest.java:47]:
```suggestion
    @Test(description = "Test case for addStringToArray() method.")
```

Check other places as well

File: src/test/java/org/wso2/securevault/secret/SecretManagerTest.java [hunk 27]
```diff
@@ -0,0 +1,156 @@
...
+            Assert.assertTrue(ex.getMessage().contains(
+                    "Invalid Annotation, The annotation expected to have [provider_type , repository_type , alias] but got [vault, admin_password]"));
+        }
+    }
+
+    private Properties getConfigProperties() {
+
+        Properties configProperties = new Properties();
+        configProperties.setProperty("keystore.identity.location",
+                "/home/sanduni/Downloads/Software Setups/wso2is-5.11.0-m28/repository/resources/security/wso2carbon" +
```
Comment (Buddhimah) [inline src/test/java/org/wso2/securevault/secret/SecretManagerTest.java:110]: This kind of location cannot be provided

Comment (Buddhimah) [reply src/test/java/org/wso2/securevault/secret/SecretManagerTest.java:110]: Check pleases where we use jks files for test cases we usually add them as test resources

File: src/main/java/org/wso2/securevault/secret/handler/SecretManagerSecretCallbackHandler.java (hunk 21 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/handler/SecretManagerSecretCallbackHandler.java:37]: Corrected

File: src/main/java/org/wso2/securevault/secret/handler/SecretManagerSecretCallbackHandler.java (hunk 22 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/handler/SecretManagerSecretCallbackHandler.java:45]: Corrected

File: src/main/java/org/wso2/securevault/secret/handler/SecretManagerSecretCallbackHandler.java (hunk 23 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/handler/SecretManagerSecretCallbackHandler.java:50]: Corrected

File: pom.xml (hunk 24 above)
Comment (sanduniNagahage) [reply pom.xml:163]: Changed

File: src/main/java/org/wso2/securevault/secret/SecretManager.java (hunk 25 above)
Comment (sanduniNagahage) [reply src/main/java/org/wso2/securevault/secret/SecretManager.java:215]: corrected

File: src/test/java/org/wso2/securevault/secret/SecretManagerTest.java (hunk 26 above)
Comment (sanduniNagahage) [reply src/test/java/org/wso2/securevault/secret/SecretManagerTest.java:47]: corrected

File: src/test/java/org/wso2/securevault/secret/SecretManagerTest.java (hunk 27 above)
Comment (sanduniNagahage) [reply src/test/java/org/wso2/securevault/secret/SecretManagerTest.java:110]: Corrected

**Comment by Buddhimah** [comment]
Closing since this has been done using a different pr https://github.com/wso2/carbon-secvault/pull/72

**Pull request closed without merging**